webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -0,0 +1,210 @@
+        .then(response=>{
+          if (response.msg === "succeeded") {
+            this.getofferList();
+            this.setState({
+              addedSucess: true,
+              conflict: false,
+              notFound: false
+            });
+          } else if (response.msg === "404") {
+            this.setState({
+              addedSucess: false,
+              notFound: true,
+              conflict: false
+            });
+          } else if (response.msg === "409") {
+            this.setState({
+              notFound: false,
+              addedSucess: false,
+              conflict: true
+            });
+          }
+        })
+      }
+    
+    displayOfferContent = (e) => {
+      const {offerList,userProfile} = this.state;
+      console.log("<<< List  Function >>>>",offerList," userProfile ",userProfile)
+     return(   
+      <div className="container"  align="center" >
+      <p className="h5 pt-5">We are pleased to offer you a place at the Deep Learning Indaba 2019. Please see the details of this offer below </p>
+
+      <form class="pt-5 ">
+        <p class="font-weight-bold">You have been accepted as a  {userProfile!=null ? userProfile.user_category: "<Category>"} </p>
+        <p class="font-weight-bold">Your status</p>
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Travel Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.travel_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Accommodation Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.accomodation_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row pb-5 ">
+          <div class="col-8 font-weight-bold">
+            Payment Required:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.payment_required
+          : "Not available"}
+          </div>
+        </div>
+        <p class="font-weight-bold">Please accept or reject this offer by {offerList!=null ? offerList.expiry_date : "<Expiry Date>"} </p>
+        <div class="row">
+         <div class="col">
+          <button type="button" class="btn btn-danger" id="reject"onClick={
+            ()=>{
+              this.setState({
+                rejected:true
+              });
+              this.buttonSubmit();
+            }
+          } >
+              Reject
+          </button>
+          </div>
+          <div class="col">
+          <button type="button" class="btn btn-success" id="accept" onClick={()=>{
+              this.setState({
+                accepted:true,
+                rejected:false,
+              });
+              this.buttonSubmit();
```

Comment:
This can cause problems because setState is asynchronous - there is no guarantee that the state has updated by the time you read it in this.buttonSubmit - see [here](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) for more information. To get around this, you can put the this.buttonSubmit inside the setState callback like this: 
```
this.setState({
                accepted:true,
                rejected:false,
              }, ()=>this.buttonSubmit());
```

Reply:
Interesting, i've checked it

Reply:
@LindanyRetroRabbit Even if you've tested it, that's only in your circumstances locally. Please follow avishkar's proposal here

Reply:
Yes, I mean't I have read the content on the... about setStaes callback and asynchronous. I also iimplemented on my code after going through the content

Reply:
@LindanyRetroRabbit You still haven't adjusted this as requested